operator/builtin/input/k8sevent/k8s_event.go

```diff
@@ -94,6 +94,23 @@ func (k *K8sEvents) Start() error {
 		return errors.Wrap(err, "build client")
 	}
 
+	if k.discoverNamespaces {
+		namespaces, err := listNamespaces(ctx, k.client)
+		if err != nil {
+			return errors.Wrap(err, "initial namespace discovery")
+		}
+		k.namespaces = append(k.namespaces, namespaces...)
```

Comment:
This looks like, if there are any namespaces specified, it will duplicate namespaces in `k.namespaces`

Reply:
Yeah, that definitely would happen. We need to take advantage of the `hasNamespace` here as well.